### PR TITLE
Lua Check Cleanup 

### LINKED ISSUE #25: Remove global variables and linting errors
With the merge of #24, potential errors in our code are now more visible:

Screenshot from 2022-01-30 21-21-30

To make them more visible in your editor please install the appropriate plugin such as [vscode-luacheck](https://marketplace.visualstudio.com/items?itemName=dwenegar.vscode-luacheck) for vscode or [ALE](https://github.com/dense-analysis/ale) for vim/neovim. This issue is resolved once we are getting :heavy_check_mark:'s instead of :x:'s by our commits.

### PR DESCRIPTION
**Pull request by wyatt-herkamp**
Replaces #27 

Closes #25 

1.  Cleaned up the Debug Menu in General

### CODE REVIEW

**Comment by wyatt-herkamp** [comment]
Screenshot from 2022-02-01 15-36-31

File: menu/menu.lua [hunk 1]
```diff
@@ -82,31 +99,32 @@ menu.draw = function()
...
 menu.update = function(dt)
-    -- Blink timer for visual effect
-    blinkTimer = blinkTimer + dt
-    if blinkTimer > 0.80 and blink == true then
-        blinkTimer = 0
-        blink = not blink
-    elseif blinkTimer > 0.2 and blink == false then
-        blinkTimer = 0
-        blink = not blink
+    -- menu.blink timer for visual effect
```
Comment (SuperShadowPlay) [inline menu/menu.lua]: This is a comment so this is very much extremely absolutely doesn't matter, but I think this is a find/replace goof

**Review by SuperShadowPlay — APPROVED** [review]
That screenshot of all the checks passing is pure serotonin.

File: menu/menu.lua (hunk 1 above)
Comment (wyatt-herkamp) [reply menu/menu.lua]: find/replace was used a lot

**Pull request merged**